File: easy/RepeatedSubstringPattern.java
public class RepeatedSubstringPattern {

    public static void main(String[] args) {
        System.out.println(repeatedSubstringPattern("abcabcabcabc"));
        // System.out.println(repeatedSubstringPattern("ababab"));
        // System.out.println(repeatedSubstringPattern("aba"));
    }

    static boolean repeatedSubstringPattern(String s) {
        int wsize = 1;
        String cur, next;
        int len = s.length();
        if (len <= 1)
            return false;
        System.err.println(len);
        // System.err.println(s.substring(0, 3)); // last isnt included
        for (int i = 0; i < len; i++) {
            System.err.println(s.charAt(i));
            cur = s.substring(i, wsize + 1);
            next = s.substring(i + wsize + 1, i + wsize + 1 + wsize);
            System.err.println("cur "+cur);
            if (cur == next) {
                System.err.println("into the if");
            }

        }

        return false;
    }
}

File: easy/RepeatedSubstringPattern.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        # start with one letter
        wsize = 1
        cur, next = None, None

        # compare it with the next
        for index in range(len(s)):
            # print(s[index], index)
            cur = s[index : index + wsize]
            next = s[index + wsize : index + wsize + 1]
            if cur != next:
                index = 0
                print("in the if", index)
                wsize += 1

            print(cur, next)

        # if valid compare with the next
        # else increase the windwsize

    # if wo


slo = Solution()

slo.repeatedSubstringPattern("ababab")
